### PR TITLE
PAAS-451 read vault servers from the db

### PR DESCRIPTION
**Pull request by grosser**
- bump vault-ruby to get `ssl_cert_store` patch
 - add UI to edit vault servers and view history
 - encrypted vault token + papertrail
 - make all vault calls retry on error
 - remove mock vault client and work with the real implementation
 - test real vault client
 - remove `vault server 0.6.2` patch ... was not needed and not working anyway after https://github.com/zendesk/samson/pull/1359

... after merge (to keep this PR manageable) solve TODOs ... migrate to vault_server_id instead of string column

### CODE REVIEW

**Comment by grosser** [comment]
screen shot 2016-10-12 at 4 46 14 pm
screen shot 2016-10-12 at 4 59 08 pm
screen shot 2016-10-12 at 4 50 07 pm
screen shot 2016-10-12 at 5 05 15 pm
screen shot 2016-10-12 at 4 51 17 pm
screen shot 2016-10-12 at 4 56 03 pm

**Comment by grosser** [comment]
@jonmoter @irwaters 
/fyi @mfischer-zd @zendesk/paas

File: lib/samson/secrets/vault_client.rb [hunk 1]
```diff
@@ -23,121 +15,83 @@ class VaultClient < Vault::Client
...
 
-      # get back the vault instance that's required for this pod.  we'll take
-      # the key, and look up the instance or return all of 'em if it's a global key
-      def vault_instances(key)
-        # parse_secret_key doesn't know about the namespace etc, so strip it off
-        deploy_group = SecretStorage.parse_secret_key(key.split('/', 3).last).fetch(:deploy_group_permalink)
-        if deploy_group == 'global'
-          @vaults.map(&:pop)
+      # local server for deploy-group specific key and all for global key
+      def clients(key)
```
Comment (jonmoter) [inline lib/samson/secrets/vault_client.rb]: Having both `client` and `clients` methods that look up different ways is a little confusing.

**Comment by grosser** [comment]
confirmed working

**Pull request merged**